pkgs/by-name/mo/moonfire-nvr/package.nix

```
@@ -1,73 +1,82 @@
 {
-  lib,
-  stdenv,
-  rustPlatform,
   fetchFromGitHub,
-  pkg-config,
+  lib,
+  moonfire-nvr,
   ncurses,
+  nodejs,
+  pkg-config,
+  pnpm,
+  rustPlatform,
   sqlite,
+  stdenv,
   testers,
-  moonfire-nvr,
-  nodejs,
-  pnpm_9,
+  tzdata,
+  ...
 }:
 
 let
   pname = "moonfire-nvr";
-  version = "0.7.20";
+  version = "0.7.21";
+
   src = fetchFromGitHub {
     owner = "scottlamb";
     repo = "moonfire-nvr";
     tag = "v${version}";
-    hash = "sha256-0EaGqZUmYGxLHcJAhlbG2wZMDiVv8U1bcTQqMx0aTo0=";
+    hash = "sha256-fkWdMFQw414en3dLzq+JulebC5GSuKIUnYIw7a92/uI=";
   };
+
   ui = stdenv.mkDerivation (finalAttrs: {
-    inherit version src;
     pname = "${pname}-ui";
-    sourceRoot = "${src.name}/ui";
+    inherit version src;
+
+    sourceRoot = "${finalAttrs.src.name}/ui";
+
     nativeBuildInputs = [
       nodejs
-      pnpm_9.configHook
+      pnpm.configHook
     ];
-    pnpmDeps = pnpm_9.fetchDeps {
-      inherit (finalAttrs) pname version src;
-      sourceRoot = "${finalAttrs.src.name}/ui";
-      hash = "sha256-7fMhUFlV5lz+A9VG8IdWoc49C2CTdLYQlEgBSBqJvtw=";
+
+    pnpmDeps = pnpm.fetchDeps {
+      inherit (finalAttrs)
+        pname
+        version
+        src
+        sourceRoot
+        ;
+
+      hash = "sha256-WY7PKVIk4ifFn6GCKYNoiCKhK1lUjwSLeC7yEDmiPWk=";
+      fetcherVersion = 9;
```

The hash have set is using fetcher version 1, and the latest supported is 2. Setting it to a higher version may result in "random" hash mismatches which are not fun to debug. Documentation: https://nixos.org/manual/nixpkgs/stable/#javascript-pnpm-fetcherVersion

(Upon merging, the package would fail to build due to https://github.com/NixOS/nixpkgs/pull/427828, as intended.)

```suggestion
      fetcherVersion = 1;
```